packages/sitecore-jss/src/site/utils.test.ts

```diff
@@ -54,6 +54,14 @@ describe('utils', () => {
       const result = getSiteRewriteData(pathname, defaultSiteName);
       expect(result.siteName).to.equal(defaultSiteName);
     });
+
+    it('should return site name from anywhere in the path', () => {
```

Comment:
We have unit test which covers this case
https://github.com/Sitecore/jss/pull/1286/files#diff-9180dcad4ff6d90d1a74047e8ed7c00e0cf601c420f20989cc514bf5faf3545aR40-R44